sqlmesh/core/engine_adapter/base.py

```diff
@@ -798,12 +798,16 @@ def _get_data_objects(
     def _get_temp_table(
         self,
         table: TableName,
+        table_only: bool = False,
     ) -> exp.Table:
         """
         Returns the name of the temp table that should be used for the given table name.
         """
         table = exp.to_table(table)
         table.set("this", f"__temp_{table.name}_{uuid.uuid4().hex}")
```

Comment:
i don't know if this table should even have db or catalog ever

Reply:
I think we need to have bigquery temp tables go in the SQLMesh schema.

Reply:
Therefore we keep the `db` in that case